vmware_aria_operations_integration_sdk/adapter_configurations/adapter_templates/java/sample_adapter/src/main/java/Adapter.java.template

```diff
@@ -1,100 +1,324 @@
+            int availableProcessors = jVMManagementInterface.getAvailableProcessors();
+            logger.info("Number of CPUs: " + availableProcessors);
+            Object cpu = result.getOrCreateObject(ADAPTER_KIND, "cpu", "CPU");
+            cpu.withProperty("cpu_count", availableProcessors);
+
+            SystemInfo systemInformation = new SystemInfo();
+
+            CentralProcessor processor = systemInformation.getHardware().getProcessor();
+
+            // Get an array of CPU tick times
+            long[] startTicks = processor.getSystemCpuLoadTicks();
+            Thread.sleep(1000);
+            long[] endTicks = processor.getSystemCpuLoadTicks();
+
+            // Calculate the CPU times
+            long userTime = endTicks[TickType.USER.getIndex()] - startTicks[TickType.USER.getIndex()];
+            long systemTime = endTicks[TickType.SYSTEM.getIndex()] - startTicks[TickType.SYSTEM.getIndex()];
+            long idleTime = endTicks[TickType.IDLE.getIndex()] - startTicks[TickType.IDLE.getIndex()];
 
-            map = mapper.readTree(inPipe);
+            cpu.withMetric("user_time", userTime);
+            cpu.withMetric("system_time", systemTime);
+            cpu.withMetric("idle_time", idleTime);
 
-        } catch (IOException e){
-                System.err.println("Error reading/writing to the pipes: " + e.getMessage());
-                e.printStackTrace();
+            // Disks
+            ArrayList<Object> disks = new ArrayList<>();
+            for (Path root : FileSystems.getDefault().getRootDirectories()) {
+                try {
+                    FileStore store = Files.getFileStore(root);
+                    Object disk = result.getOrCreateObject(ADAPTER_KIND, "disk", "Disk");
+                    disk.withProperty("partition", root.toString());
+
+                    // Disk stats
+                    long totalSpace = store.getTotalSpace();
+                    long freeSpace = store.getUsableSpace();
+                    long usedSpace = totalSpace - freeSpace;
+                    double utilization = (double) usedSpace / totalSpace * 100;
+
+                    // Adding metrics to disk
+                    disk.withMetric("total_space",totalSpace);
+                    disk.withMetric("free_space", freeSpace);
+                    disk.withMetric("used_space", freeSpace);
+                    disk.withMetric("disk_utilization",utilization);
+
+                    if (utilization >= 90 ){
+                        disk.withEvent("High disk utilization");
+                    }
+
+                    disks.add(disk);
+
+                } catch (IOException e) {
+                    logger.error("Unable to process disk data due to: " + e.getMessage());
+                    logger.debug(e.getStackTrace());
+                }
+            }
+
+            // Add System Object
+            Object system = result.getOrCreateObject(ADAPTER_KIND, "system", "System");
```

Comment:
Might be worthwhile to set the name of the system object to the `id` parameter, as an example.